### PR TITLE
generator: fail when no exercises matched the given options

### PR DESCRIPTION
**Pull request by jrr**
Here's an extremely simple solution to https://github.com/exercism/fsharp/issues/621

I don't know if anyone or anything was relying on the existing _generate-nothing-and-throw-no-errors_ behavior, but thought I'd start the discussion and try to improve the experience for other people that land here via Hacktoberfest. I'm totally open to other, better ways of improving this!

### CODE REVIEW

File: generators/Program.fs [hunk 1]
```diff
@@ -49,6 +49,9 @@ let private regenerateTestClasses options =
     
     createExercises options
     |> List.filter (shouldBeIncluded options)
+    |> (function
```
Comment (ErikSchierboom) [inline generators/Program.fs]: I really like this feature! Well thought of!

I might prefer a little change to the implementation though. Perhaps you could extract the logic to a helper function, in which you can then use the `Log.Error` function to log the actual error. This is more consistent with how errors are handled elsewhere in the file.

Another option would be to do some matching:

```fsharp
let exercisesToGenerate = createExercises options |> List.filter (shouldBeIncluded options)
match exercisesToGenerate with
| [] -> Log.Error("No exercises matched given options")
| _  -> List.iter regenerateTestClass' exercisesToGenerate
```

See which ever you prefer.

File: generators/Program.fs [hunk 2]
```diff
@@ -49,9 +49,11 @@ let private regenerateTestClasses options =
     
     createExercises options
     |> List.filter (shouldBeIncluded options)
-    |> List.iter regenerateTestClass'
-
-    Log.Information("Re-generated test classes.")
+    |> function
+        | [] -> Log.Error "No exercises matched given options."
+        | x ->
```
Comment (ErikSchierboom) [inline generators/Program.fs]: Tiny nit: could you rename `x` to `exercises`?

File: generators/Program.fs (hunk 1 above)
Comment (jrr) [reply generators/Program.fs]: Alright, I pushed an update that's neither of those things.

I like avoiding the _"Re-generated test classes."_ message when nothing was generated, and I almost always prefer extending pipelines over declaring intermediates.

What do you think? Do you have a strong preference for one of your suggestions over this? Totally willing to change it again.

File: generators/Program.fs (hunk 2 above)
Comment (jrr) [reply generators/Program.fs]: Good idea! I could stand to develop a habit of naming those better.

File: generators/Program.fs [hunk 3]
```diff
@@ -49,9 +49,11 @@ let private regenerateTestClasses options =
     
     createExercises options
     |> List.filter (shouldBeIncluded options)
-    |> List.iter regenerateTestClass'
-
-    Log.Information("Re-generated test classes.")
+    |> function
+        | [] -> Log.Error "No exercises matched given options."
```
Comment (ErikSchierboom) [inline generators/Program.fs]: Maybe `Log.Warning` would be a better error level on second thought. So sorry about this.

**Comment by ErikSchierboom** [comment]
Love it! Thanks a lot. 🎉

**Pull request merged**